test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -334,13 +334,20 @@ export default class PerfCompare extends Component {
         } else {
 
             const [baselineSchemeWithHostWithPort, baselineBuildNum] = this.state.inputURL.baselineID.split("build_id=");
-            const [testSchemeWithHostWithPort, testBuildNum] = this.state.inputURL.testID.split("build_id=");
-
+            const [testSchemeWithHostWithPort, testBuildNum] = this.state.inputURL.testID.split("build_id=");   
+            let server_2 = await fetch( `/api/getPerffarmRunCSV?${stringify(this.state.selectedRuns)}`, {method : 'get'});
+			const server_3 = await server_2.json();
+			const server = server_3.serverId;
+                                     
             // Check if the benchmark and test data is valid
-            if (baselineBuildNum === undefined || (baselineSchemeWithHostWithPort.substring(0,31) !== "http://perffarmServer")) {
+            if ((baselineSchemeWithHostWithPort.substring(0, server.length) !== server)){
+           		 displayErrorMessage += "Invalid ServerID URL."
+           }
+            
```

Comment:
Instead of validating user-provided server url here, could we just store it in the state first and valid it later with the rest info on the server side `getPerffarmRunCSV.js`?

`selectedRuns` in the state should look like following:

```
selectedRuns: {
                        serverUrl: xxx,
                        baselineID: baselineBuildNum,
                        testID: testBuildNum,
                    }	
```

Reply:
@llxia @piyush286 I updated these two points, would you please review the changes. Thank you.

Reply:
We have an API call (line 338) to get expected server url and validate it. And have the same API call (line 93) again to pass in `baselineBuildNum` and `testBuildNum`. This is not necessary. I would like to just use one API call.

Instead of line338-line347, could we just get serverURL and do some preliminary/simple checks (i.e., not empty), set serverURL in state, then at line 93 `getPerffarmRunCSV` gets called and `getPerffarmRunCSV` can validate `serverUrl` as well as `baselineBuildNum` and `testBuildNum`?

Reply:
@llxia I think it would be useful to show this error (i.e. line 345), hence don't we need some check like this on the client side?
`if (serverURL !== server){
         displayErrorMessage += "Invalid ServerID URL."
}`

Earlier, when the serverURL wasn't set correctly (i.e. set to * http://perffarmServer), the user would get this error "Invalid Baseline URL. Invalid Test URL.", which is deceiving since that's not the real issue. 

I agree that having 2 calls to `getPerffarmRunCSV` might be a little redundant.

Reply:
Server-side API can return whatever message we think is reasonable. It can return `Invalid ServerID URL` if needed.

Reply:
Now I understand what you meant! Yeah, that would definitely be a cleaner approach. Thanks!